buildpipeline/perf-pipeline.groovy

```diff
@@ -0,0 +1,390 @@
+    archiveArtifacts allowEmptyArchive: false, artifacts:'throughput-*.csv,machinedata.json'
+}
+
+//------------------------ Linux Functions ----------------------------//
+
+def linuxBuild(String arch, String config, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String pgoStash = ((pgo == "skiprestoreoptdata") ? "nopgo" : "pgo")
+    String baselineString = ""
+
+    // For baseline runs, checkout the merge's parent
+    if (isBaseline) {
+        baselineString = "-baseline"
+        sh "git checkout HEAD^1"
+    }
+
+    sh "./build.sh verbose ${config} ${arch} ${pgo}"
+    stash name: "linux-${arch}-${pgoStash}${baselineString}-build-artifacts", includes: 'bin/**'
+}
+
+def linuxPerf(String arch, String os, String config, String uploadString, String runType, String optLevel, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String baselineString = ""
+    if (isBaseline) {
+        baselineString = "-baseline"
+    }
+
+    dir ('.') {
+        unstash "linux-${arch}-${pgo}${baselineString}-build-artifacts"
+        unstash "nt-${arch}-${pgo}${baselineString}-test-artifacts"
+        unstash "benchview-artifacts"
+        unstash "metadata"
+    }
+
+    // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+    // submission-metadata.py
+    if (isBaseline) {
+        sh "mv submission-metadata-baseline.json submission-metadata.json"
+    }
+
+    sh "./tests/scripts/perf-prep.sh"
+    sh "./init-tools.sh"
+    sh "./tests/scripts/run-xunit-perf.sh --testRootDir=\"\${WORKSPACE}/bin/tests/Windows_NT.${arch}.${config}\" --testNativeBinDir=\"\${WORKSPACE}/bin/obj/Linux.${arch}.${config}/tests\" --coreClrBinDir=\"\${WORKSPACE}/bin/Product/Linux.${arch}.${config}\" --mscorlibDir=\"\${WORKSPACE}/bin/Product/Linux.${arch}.${config}\" --coreFxBinDir=\"\${WORKSPACE}/corefx\" --runType=\"${runType}\" --benchViewOS=\"${os}\" --stabilityPrefix=\"taskset 0x00000002 nice --adjustment=-10\" --uploadToBenchview --generatebenchviewdata=\"\${WORKSPACE}/tests/scripts/Microsoft.BenchView.JSONFormat/tools\""
+    archiveArtifacts allowEmptyArchive: false, artifacts:'Perf-*.xml,Perf-*.log,machinedata.json'
+}
+
+def linuxThroughput(String arch, String os, String config, String uploadString, String runType, String optLevel, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String baselineString = ""
+    if (isBaseline) {
+        baselineString = "-baseline"
+    }
+
+    dir ('.') {
+        unstash "linux-${arch}-${pgo}${baselineString}-build-artifacts"
+        unstash "benchview-artifacts"
+        unstash "throughput-benchmarks-${arch}"
+        unstash "metadata"
+    }
+
+    // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+    // submission-metadata.py
+    if (isBaseline) {
+        sh "mv submission-metadata-baseline.json submission-metadata.json"
```

Comment:
To mirror Windows's `move /y`, I think we want `mv -f` here

Reply:
Good catch.